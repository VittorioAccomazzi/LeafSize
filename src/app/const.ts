export const root = "/LeafSize/";
export const settingsPath  = `${root}settings`;
export const selectionPath = `${root}`;
export const processingPath= `${root}process`;
export const resultPath    = `${root}result`;
