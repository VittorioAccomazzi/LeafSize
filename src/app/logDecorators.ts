

//
// Method decorators
//
export default  function timeTaken  (target: any, propertyKey: string, descriptor: PropertyDescriptor) {
    const originalMethod = descriptor.value;

    descriptor.value = function (...args: any) {
      const start = Date.now();
      const result = originalMethod.apply(this, args);
      const finish = Date.now();
      console.log(`${target.name}.${propertyKey}: ${finish - start} ms`);
      return result;
    };
  
    return descriptor;
  }

  export function logParameters (target: any, propertyKey: string, descriptor: PropertyDescriptor) {
    const originalMethod = descriptor.value;

    descriptor.value = function (...args: any[]) {
      console.log(`${target.name}.${propertyKey}: ${args.length} parameters :`);
      for( let [key,val] of Object.entries(args) ){
          console.log(`  ${key} : ${val}`)
      }
      const result = originalMethod.apply(this, args);
      return result;
    };
  
    return descriptor;
  }

//
//  Class Decorator
//
export function addToString(target: any) {
    target.prototype.toString = () : string => {
        return `object of type ${target.name}`
    }
    return target
}
