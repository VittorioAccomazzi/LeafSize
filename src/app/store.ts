import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import selectionReducer from '../pages/selection/selectionSlice';
import settingsReducer from '../pages/settings/settingSlice'

export const store = configureStore({
  reducer: {
    selection : selectionReducer,
    settings : settingsReducer
  },
  // 🖐 The following line are necessary since we store FileSystemHandle and FileSystemDirectoryHandle
  // in teh Redux store. These classes are not serializable by javascript right now, but they are by
  // indexDB and postmessage. So I'm wondering if this can be extended ....
  // see my question on https://github.com/WICG/file-system-access/issues/368 in order to avoid this.
  middleware: (getDefaultMiddleware) =>
  getDefaultMiddleware({
    serializableCheck: {
      // Ignore these action types
      ignoredActions: ['selection/setFolder','selection/setFiles'],
      // Ignore these field paths in all actions
      ignoredActionPaths: ['payload.0'],
      // Ignore these paths in the state
      ignoredPaths: ['selection.folder', 'selection.files'],
    }, 
  }),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
