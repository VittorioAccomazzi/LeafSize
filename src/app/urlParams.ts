
/**
 * Simple class which centralize the access to the URL parameters for the application
 */
export default class URLParams {

    private static params = new URLSearchParams( typeof window === 'undefined' ? '' : window.location.search );
    private constructor() { } // singleton

    /**
     * for web web wrokers
     * @param params 
     */
    static SetSearchParam( urlParams: string ){
        URLParams.params = new URLSearchParams(urlParams); 
    }

    /**
     * return the pack mode for the pathogen segmentation
     */
    static get PackMode () : string  {
        return URLParams.params.get('PackMode') ?? "Lab";
    }

}