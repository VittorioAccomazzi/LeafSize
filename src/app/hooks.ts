import { useEffect, useMemo } from 'react';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { ImageFinalized, selectFinalized } from '../pages/process/ProcessSlice';
import { root } from './const';
import type { RootState, AppDispatch } from './store';

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

// redirect to selection page if there are no files selected.
export const useAutomaticRedirect = (files : FileSystemFileHandle []|null ) => {
    const navigate = useNavigate();
    useEffect(()=> {
        if(!(files?.length)) navigate(root)
    }, []) // eslint-disable-line  react-hooks/exhaustive-deps -- shall ran only once.
}


// simple hook to select the images which still need processing from the store.

const searchElement = ( list : ImageFinalized [], name : string ) : boolean => list.findIndex(el=>el.name===name) >= 0
export const  useImagesToProcess = ( imageList : string [] ) => {
    const imgFinalized = useAppSelector(selectFinalized);
    const imagesToProcess = useMemo(()=>imageList.filter( v => !searchElement(imgFinalized,v) ) , []); // eslint-disable-line  react-hooks/exhaustive-deps -- on purpose this shall be computed only once.
    return imagesToProcess;
}
