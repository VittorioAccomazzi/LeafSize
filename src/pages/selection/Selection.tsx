import { useState } from "react";
import { Box, Button, Stack, Typography } from "@mui/material";
import { setNumDishes, setNumLeafs, setFolder, setFiles }  from './selectionSlice';
import { selectNumDishes, selectNumLeafs, selectFolder, selectFiles } from "./selectionSlice";
import { NextPage } from "../../common/components/NavButtons";
import { useAppDispatch, useAppSelector } from "../../app/hooks";
import dish4Leaf1  from '../../assets/dish-4-leaf-1.svg';
import dish1Leaf1  from '../../assets/dish-1-leaf-1.svg';
import dish1Leaf2  from '../../assets/dish-1-leaf-2.svg';
import OptionsList from "../../common/components/OptionList";
import isImage from "../../common/utils/FileUtils";


export default function Selection() {
    const folder = useAppSelector(selectFolder);
    const files  = useAppSelector(selectFiles);
    const numDishes = useAppSelector(selectNumDishes);
    const numLeafs  = useAppSelector(selectNumLeafs);
    const [loading, setLoading] = useState<boolean>(false)
    const dispatch  = useAppDispatch();

    const setImgFolder = async () =>{
        const dirHandle = await showDirectoryPicker();
        const files : FileSystemFileHandle[] = []
        setLoading(true);
        for await (const entry of dirHandle.values()) {
          if( entry.kind === 'file' && isImage( entry.name) ) files.push(entry)
        }
        dispatch(setFolder(dirHandle));
        dispatch(setFiles(files));
        setLoading(false);
    }
    return (
        <>
            <Typography color="text.primary" variant="h2" marginBottom="1em">Select folder and images layout</Typography>
            <Box marginBottom="1em" alignItems='center' display='flex' flexDirection='column'>
                <Stack spacing={4} >
                    <Stack spacing={2} direction="row" alignItems="baseline">
                        <Button variant="outlined" onClick={setImgFolder} disabled={loading}> Select Folder</Button> 
                        <Typography minWidth="12em">{folder ? folder.name : null }</Typography>
                        <Typography color="text.secondary" minWidth="12em">{files ? `(${files.length} files selected)` : ''} </Typography>
                    </Stack>
                    <OptionsList
                        imgSrc1={dish1Leaf1}
                        imgSrc2={dish4Leaf1}
                        label1="One Dish"
                        label2="Four Dishes"
                        selected1 = {numDishes===1}
                        selected2 = {numDishes===4}
                        click1={()=>dispatch(setNumDishes(1))}
                        click2={()=>dispatch(setNumDishes(4))}
                        title="Select Image Layout"
                    />
                    <OptionsList
                        imgSrc1={dish1Leaf1}
                        imgSrc2={dish1Leaf2}
                        label1="One leaf per dish"
                        label2="Two leafs per dish"
                        selected1={numLeafs===1}
                        selected2={numLeafs===2}
                        click1={()=>dispatch(setNumLeafs(1))}
                        click2={()=>dispatch(setNumLeafs(2))}
                        title="Select Number of leafs per dish" />
                </Stack>   
            </Box>
            <NextPage page="/settings" disabled={loading || files==null || files.length===0}/>
        </>
    )
}
