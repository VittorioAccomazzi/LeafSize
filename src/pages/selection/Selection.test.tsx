import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../app/store';
import Selection from './Selection';
import { BrowserRouter, Route, Routes } from 'react-router-dom';


test('Selection', () => {
    const wrapper = render(
      <Provider store={store}>
        <BrowserRouter>
            <Routes>
                <Route path='/' element={<Selection/>}/>
            </Routes>
        </BrowserRouter>
      </Provider>
    );

    expect(wrapper.getByText('Select folder and images layout')).toBeInTheDocument();
  });
