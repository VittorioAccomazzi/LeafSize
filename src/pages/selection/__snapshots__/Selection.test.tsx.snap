// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selection 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="MuiTypography-root MuiTypography-h2 css-qkjx0-MuiTypography-root"
      >
        Select folder and images layout
      </h2>
      <div
        class="MuiBox-root css-14icbzs"
      >
        <div
          class="css-678bp8-MuiStack-root"
        >
          <div
            class="css-q2d8kl-MuiStack-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
               Select Folder
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <p
              class="MuiTypography-root MuiTypography-body1 css-9ctv4h-MuiTypography-root"
            />
            <p
              class="MuiTypography-root MuiTypography-body1 css-2jqf5p-MuiTypography-root"
            >
               
            </p>
          </div>
          <div
            class="css-p6dskh-MuiStack-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
            >
              Select Image Layout
            </label>
            <div
              class="css-1mkkg5w-MuiStack-root"
            >
              <span
                class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  type="radio"
                  value=""
                />
                <span
                  class="css-hyxlzm"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <img
                height="32px"
                src="dish-1-leaf-1.svg"
              />
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
              >
                One Dish
              </label>
            </div>
            <div
              class="css-1mkkg5w-MuiStack-root"
            >
              <span
                class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root Mui-checked css-vqmohf-MuiButtonBase-root-MuiRadio-root"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  type="radio"
                  value=""
                />
                <span
                  class="css-hyxlzm"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <img
                height="32px"
                src="dish-4-leaf-1.svg"
              />
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
              >
                Four Dishes
              </label>
            </div>
          </div>
          <div
            class="css-p6dskh-MuiStack-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
            >
              Select Number of leafs per dish
            </label>
            <div
              class="css-1mkkg5w-MuiStack-root"
            >
              <span
                class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  type="radio"
                  value=""
                />
                <span
                  class="css-hyxlzm"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <img
                height="32px"
                src="dish-1-leaf-1.svg"
              />
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
              >
                One leaf per dish
              </label>
            </div>
            <div
              class="css-1mkkg5w-MuiStack-root"
            >
              <span
                class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root Mui-checked css-vqmohf-MuiButtonBase-root-MuiRadio-root"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  type="radio"
                  value=""
                />
                <span
                  class="css-hyxlzm"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <img
                height="32px"
                src="dish-1-leaf-2.svg"
              />
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
              >
                Two leafs per dish
              </label>
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-1fej5jl-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <div
          class="css-7q56u8-MuiStack-root"
        >
          Next
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1wi0ike-MuiSvgIcon-root"
            data-testid="NavigateNextIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
           
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <h2
      class="MuiTypography-root MuiTypography-h2 css-qkjx0-MuiTypography-root"
    >
      Select folder and images layout
    </h2>
    <div
      class="MuiBox-root css-14icbzs"
    >
      <div
        class="css-678bp8-MuiStack-root"
      >
        <div
          class="css-q2d8kl-MuiStack-root"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
             Select Folder
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <p
            class="MuiTypography-root MuiTypography-body1 css-9ctv4h-MuiTypography-root"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 css-2jqf5p-MuiTypography-root"
          >
             
          </p>
        </div>
        <div
          class="css-p6dskh-MuiStack-root"
        >
          <label
            class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
          >
            Select Image Layout
          </label>
          <div
            class="css-1mkkg5w-MuiStack-root"
          >
            <span
              class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                type="radio"
                value=""
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <img
              height="32px"
              src="dish-1-leaf-1.svg"
            />
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
            >
              One Dish
            </label>
          </div>
          <div
            class="css-1mkkg5w-MuiStack-root"
          >
            <span
              class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root Mui-checked css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                type="radio"
                value=""
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <img
              height="32px"
              src="dish-4-leaf-1.svg"
            />
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
            >
              Four Dishes
            </label>
          </div>
        </div>
        <div
          class="css-p6dskh-MuiStack-root"
        >
          <label
            class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
          >
            Select Number of leafs per dish
          </label>
          <div
            class="css-1mkkg5w-MuiStack-root"
          >
            <span
              class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                type="radio"
                value=""
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <img
              height="32px"
              src="dish-1-leaf-1.svg"
            />
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
            >
              One leaf per dish
            </label>
          </div>
          <div
            class="css-1mkkg5w-MuiStack-root"
          >
            <span
              class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root Mui-checked css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                type="radio"
                value=""
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <img
              height="32px"
              src="dish-1-leaf-2.svg"
            />
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
            >
              Two leafs per dish
            </label>
          </div>
        </div>
      </div>
    </div>
    <button
      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-1fej5jl-MuiButtonBase-root-MuiButton-root"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <div
        class="css-7q56u8-MuiStack-root"
      >
        Next
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1wi0ike-MuiSvgIcon-root"
          data-testid="NavigateNextIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
         
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
